linc_convert/modalities/lsm/spool.py

```diff
@@ -0,0 +1,283 @@
+
+    if isinstance(compressor_opt, str):
+        compressor_opt = ast.literal_eval(compressor_opt)
+
+    CHUNK_PATTERN = re.compile(
+        r"^(?P<prefix>\w*)"
+        r"_run(?P<run>[0-9]+)"
+        r"_?"
+        r"_y(?P<y>[0-9]+)"
+        r"_z(?P<z>[0-9]+)"
+        r"(?P<suffix>\w*)$"
+    )
+
+    all_tiles_folders_names = sorted(glob(os.path.join(inp, "*_y*_z*_HR/")))
+    if not all_tiles_folders_names:
+        raise ValueError("No tile folder found")
+
+    all_tiles_info = []
+    tiles_info_by_index = {}
+    TileInfo = namedtuple(
+        "TileInfo",
+        ["prefix", "run", "y", "z", "suffix", "filename", "reader"]
+    )
+    for tile_folder_name in all_tiles_folders_names:
+        tile_folder_name = tile_folder_name.rstrip("/")
+        parsed = CHUNK_PATTERN.fullmatch(os.path.basename(tile_folder_name))
+        tile = TileInfo(
+            parsed.group("prefix"),
+            int(parsed.group("run")),
+            int(parsed.group("y")),
+            int(parsed.group("z")),
+            parsed.group("suffix"),
+            tile_folder_name,
+            SpoolSetInterpreter(tile_folder_name, tile_folder_name+"_info.mat")
+        )
+        all_tiles_info.append(tile)
+        # Check for duplicate tiles
+        if (tile.y, tile.z) in tiles_info_by_index:
+            raise ValueError(
+                f"Duplicate tile, file {tile.filename} conflicts with "
+                f"{tiles_info_by_index[(tile.y, tile.z)].filename}"
+            )
+        tiles_info_by_index[(tile.y, tile.z)] = tile
+
+    # Set default output path if not provided
+    if not out:
+        out = all_tiles_info[0].prefix + all_tiles_info[0].suffix
+        out += ".nii.zarr" if nii else ".ome.zarr"
+    nii = nii or out.endswith(".nii.zarr")
+
+    # Determine unique Y and Z tile indices
+    z_tiles = set(tile.z for tile in all_tiles_info)
+    y_tiles = set(tile.y for tile in all_tiles_info)
+    min_y_tile, max_y_tile = min(y_tiles), max(y_tiles)
+    min_z_tile, max_z_tile = min(z_tiles), max(z_tiles)
+    num_y_tiles, num_z_tiles = len(y_tiles), len(z_tiles)
+
+    # Initialize dtype and shapes storage
+    dtypes = defaultdict(list)
+    expected_sx = 0
+    expected_sy = {}
+    expected_sz = {}
+    all_shapes = np.empty((num_y_tiles, num_z_tiles, 3), dtype=int)
+
+    # Collect shape and dtype info from all tiles
+    for z_tile in range(min_z_tile, max_z_tile + 1):
+        for y_tile in range(min_y_tile, max_y_tile + 1):
+            tile = tiles_info_by_index.get((y_tile, z_tile))
+            if tile is None:
+                warnings.warn(f"Missing tile {y_tile}, {z_tile}")
+                continue
+            reader = tile.reader
+            sz, sy, sx = reader.assembled_spool_shape
+            dt = reader.dtype
+            # Collect shapes and dtypes.
+            rel_y, rel_z = y_tile - min_y_tile, z_tile - min_z_tile
+            all_shapes[rel_y, rel_z] = sx, sy, sz
+            expected_sx = sx
```

Comment:
I might be misunderstanding, but do we want to check that this value is the same for all tiles?  It looks like this value is being overwritten as the for loops iterate.